src/asciidoc/docs/develop/testingTemplates.adoc

```diff
@@ -0,0 +1,208 @@
+
+=== Testing backend routes
+
+The *Routes* tab lets you test your backend routes.
+It's similar to using a program like curl or Fiddler.
+For example, you can manually enter an HTTP request, configure a header for it, send the request, and examine the results returned by the server.
+
+=== Writing to the console
+
+You can test JavaScript code by using the `console.log` method, which prints to the console.
+To see the console in the Koji editor, click the console button image:consoleButton.jpg[Console button] in the *Live preview* pane.
+
+For example, the following code snippet prints *Key pressed* to the console every time you press a key, as long as the *Live preview* pane is selected.
+
+[NOTE]
+You can select *Live preview* by clicking anywhere in its pane.
+
+[source,javascript]
+----
+    $(window).keydown(function (e) {
+        KEY_STATUS.keyDown = true;
+        if (KEY_CODES[e.keyCode]) {
+            e.preventDefault();
+            KEY_STATUS[KEY_CODES[e.keyCode]] = true;
+        }
+        console.log("Key pressed");
+    }).keyup(function (e) {
+        KEY_STATUS.keyDown = false;
+        if (KEY_CODES[e.keyCode]) {
+            e.preventDefault();
+            KEY_STATUS[KEY_CODES[e.keyCode]] = false;
+        }
+    });
+----
+
+When running your Koji outside the editor, you can access the console through the browser's menu.
+(See <<_using_browser_tools_for_testing>>.)
+
+=== Testing in-app purchases
+
+If your template supports <<withkoji-koji-iap-package#,in-app purchases>>, you can test the functionality by clicking *Debug in live player* on the *Remote* tab.
+
+. Log into your Koji account and click your profile icon to open the main menu.
+. Click *Wallet*.
+. If your *Wallet* is empty, top it up.
+. In another browser window, open the Koji editor.
+. Open the *Remote* tab and click *Debug in live player*.
+. Try paying yourself a small amount.
+. Go back to your *Wallet* to check that the transaction was successful.
+. If all went well, you'll see the amount you paid deducted from your balance.
+The transaction will be marked *pending*.
+. You should be able to refund the transaction, resetting your balance to its original value and cancelling the transaction fee.
++
+[NOTE]
+If *KojiPay* doesn't allow you to refund the transaction, it might not be finished processing the payment.
+Wait a minute or two and try again.
+
+=== Testing a social media share
+
+When a user shares a remix of your template on social media, the Koji platform renders a rich preview (Open Graph) image based on a screenshot of the Koji.
+You can preview this image by opening the *Thumbnail* tab in the *Live preview* pane.
+For information about how to customize the image, see <<rich-preview-image#>>.
+
+=== Viewing messages passed between a template and platform APIs
+
+Platform features like VCCs and in-app purchases require communication between your Koji and platform APIs.
+
```

Comment:
It looks like this extra line was added to separate this section into two paragraphs. Was that intentional? It could work either way so up to you whether to keep it.

Reply:
Probably works better without the extra line.